gateway/gateway.go

```diff
@@ -166,7 +154,7 @@ type gateway struct {
 	lastReadings map[string]interface{} // map of devices to readings
 	readingsMu   sync.Mutex
 
-	devices map[string]*node.Node // map of node name to node struct
+	devices map[string]gatewayNode // map of node name to node struct
```

Comment:
the initial change was to simplify how `devices` was being used and to make it safe. now that we want to move to sqlite this change isn't super relevant anymore, but I still wanted to decouple the node struct from the gateway logic as it was causing some confusion